typed/remove_test.go

```diff
@@ -0,0 +1,457 @@
+		{
+			object:          `{"a": "value", "b": [{"c": "value"}, {"d": "value"}]}`,
+			remove:          _NS(_P("b", _I(0), "c")),
+			expect:          `{"a": "value", "b": [{}, {"d": "value"}]}`,
+			expectItemsOnly: `{"a": "value", "b": [{}, {"d": "value"}]}`,
+		},
+		{
+			object:          `{"a": "value", "b": {"c":"value", "d":{"e":"value"}}}`,
+			remove:          _NS(_P("b", "d")),
+			expect:          `{"a": "value", "b": {"c":"value"}}`,
+			expectItemsOnly: `{"a": "value", "b": {"c":"value"}}`,
+		},
+		{
+			object:          `{"a": "value", "b": {"c":"value", "d": {"e":"value"}}}`,
+			remove:          _NS(_P("b", "d", "e")),
+			expect:          `{"a": "value", "b": {"c":"value", "d": {}}}`,
+			expectItemsOnly: `{"a": "value", "b": {"c":"value", "d": {}}}`,
+		},
+	}
+
+	for i, c := range cases {
+		c := c
+		t.Run(fmt.Sprintf("case-%d", i), func(t *testing.T) {
+			t.Parallel()
+
+			obj, err := DeducedParseableType.FromYAML(c.object)
+			if err != nil {
+				t.Fatalf("unable to parse/validate object: %v\n%v", err, c.object)
+			}
+
+			parseable := &ParseableType{
+				Schema:  obj.schema,
+				TypeRef: obj.typeRef,
+			}
+			expectTyped, err := parseable.FromYAML(c.expect)
+			if err != nil {
+				t.Fatalf("unable to parse/validate expected object: %v\n%v", err, c.expect)
+			}
+			expect := expectTyped.AsValue()
+
+			expectItemsOnlyTyped, err := parseable.FromYAML(c.expectItemsOnly)
+			if err != nil {
+				t.Fatalf("unable to parse/validate expected object: %v\n%v", err, c.expectItemsOnly)
+			}
+			expectItemsOnly := expectItemsOnlyTyped.AsValue()
+
+			result := obj.Remove(c.remove).AsValue()
+			if !value.Equals(result, expect) {
+				t.Fatalf("unexpected result after Remove:\ngot: %v\nexp: %v",
+					value.ToString(result), value.ToString(expect),
+				)
+			}
+
+			result = obj.RemoveItems(c.remove).AsValue()
+			if !value.Equals(result, expectItemsOnly) {
+				t.Fatalf("unexpected result after RemoveItems:\ngot: %v\nexp: %v",
+					value.ToString(result), value.ToString(expectItemsOnly),
+				)
+			}
+		})
+	}
+}
+func TestRemove(t *testing.T) {
+	var cases = []struct {
+		object          YAMLObject
+		schema          YAMLObject
+		remove          *fieldpath.Set
+		expect          YAMLObject
+		expectItemsOnly YAMLObject
+	}{
+		{
+			object: `{"a": "value", "b":{"c":"value"}}`,
+			schema: `types:
+        - name: type
+          map:
+            fields:
+              - name: a
+                type:
+                  scalar: string
+              - name: b
+                type:
+                  map:
+                    fields:
+                    - name: c
+                      type:
+                        scalar: string
+            elementType:
+              scalar: string
+        `,
+			remove:          _NS(_P("b")),
+			expect:          `{"a": "value"}`,
+			expectItemsOnly: `{"a": "value", "b":{"c":"value"}}`,
+		},
+		{
+			object: `{"a": "value", "b":{"c":"value"}}`,
+			schema: `types:
+- name: type
```

Comment:
somehow this one is not indented the same way as the other

Reply:
it's not the only one either.

Reply:
Yeah, missed that.
I commented the other tests for debugging only broken ones and removing the comments messed up indendation.